frontend/src/components/RestaurantList/TagsTab.js

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {withStyles} from '@material-ui/core/styles';
+import AppBar from '@material-ui/core/AppBar';
+import Tabs from '@material-ui/core/Tabs';
+import Tab from '@material-ui/core/Tab';
+import Typography from '@material-ui/core/Typography';
+import RestaurantItem from './RestaurantItem';
+import {Link} from "react-router-dom";
+
+function TabContainer(props) {
+    return (
+        <Typography component="div">
+            {props.children}
+        </Typography>
+    );
+}
+
+TabContainer.propTypes = {
+    children: PropTypes.node.isRequired,
+};
+
+const styles = theme => ({
+    root: {
+        flexGrow: 1,
+        backgroundColor: theme.palette.background.paper,
+    },
+    appbar: {
+        backgroundColor: 'rgba(230, 232, 209, 0.96)',
+        color: "black",
+        borderRadius: 7,
+    },
+    tabs: {
+        fontSize: 20,
+        fontWeight: 600
+    }
+});
+
+class TagsTab extends React.Component {
+    state = {
+        value: 0,
+        rests: [],
+    };
+
+    componentDidMount() {
+        fetch('http://localhost:3000/restaurant.json')
+            .then(response => response.json())
+            .then(data => this.setState({rests: data.data}))
+            .catch(err=>console.log(err));
+
+    }
+
+    handleChange = (event, value) => {
+        this.setState({value});
+    };
+
+    handleGetTags=( rests)=>{
+        let tagNames=[];
+        rests.map(item=> item.tags.map(tag=>{
+            if (tagNames.indexOf(tag.name) === -1)
+                tagNames.push(tag.name);
+            return '';
+        }));
+        return tagNames;
+    };
+
+    render() {
+        const {classes} = this.props;
+        let {value} = this.state;
+        const tagNames=this.handleGetTags(this.state.rests);
+        let searchTag =this.props.url.location.search;
+        let params = new URLSearchParams(this.props.url.location.search);
+
+
+        if (params.get("tag") !== null){
+            value=params.get("tag");
+        } else value=0;
```

Comment:
1) DO NOT CHANGE STATE INSIDE RENDER method like this!!!!!!!
In React `render()` method called in two general cases: state changed or props changed. If you change state inside `render()` method you can cause recursion: State changed -> render() called -> state changed -> render() called.....
2) If you want change state ALWAYSE use `setState` method!!!!! -> read the docs why

Reply:
It's ok in this case as `value` here is just a copy of the attribute from the state. But it's a bit strange to init `value` from state, but don't use it later. As here `value` will be overwritten in both cases.

Reply:
If I want change state, I ALWAYSE use setState method!!!!!)) 
I do not change the state, I forgot to remove the value from the state